### PR TITLE
Smart contracts: add Shuffle method to the Partition interface:

### LINKED ISSUE #1050: Randomly shuffling partitions
To  prevent Byzantine generation attacks we want to randomly shuffle partitions as used in blobber block rewards and challenge generation.

We add a shuffle partition item method to the [Partition interface](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/partitions/partition.go#L33)
1. We have a blobber Bk in Partition section Pk.
2. Randomly select a partition section Pi.
3. Swap the blobber Bkn in Pk with the a blobber Bin in partition section Pi.
4. Update B and B1 so they point to their new locations in the partition.

We can shuffle a blobber in the block rewards and challenge generation partitions during [verifyChallenge](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/challenge.go#L319).

### PR DESCRIPTION
**Pull request by MurashovVen**
- Implementing Shuffle by the randomSelector
- Add set and getByIndex methods to the PartitionItemList interface and implementations

Closes #1050

### CODE REVIEW

File: code/go/0chain.net/smartcontract/partitions/random.go [hunk 1]
```diff
@@ -200,6 +198,67 @@ func (rs *randomSelector) GetRandomSlice(
...
+
+	case !(partitionB >= 0 && partitionB < len(rs.Partitions)):
+		return fmt.Errorf("partition B: %w", IndexOutOfBounds)
+	}
+
+	a, err := rs.Partitions[partitionA].getByIndex(itemA)
+	if err != nil {
+		return fmt.Errorf("can't get item A from partition: %w", err)
+	}
+	a = a.Copy()
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/partitions/random.go:244]: why do we need to operate with copies of a, b

**Comment by peterlimg** [comment]
@MurashovVen please resolve the conflicts

**Comment by peterlimg** [comment]
@MurashovVen please resolve the conflicts

**Comment by dabasov** [comment]
not needed anymore, issue is closed

**Pull request closed without merging**